### PR TITLE
[server] Update doc path for server

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

Currently, documentation is based on file instead of directory for the server. 

See https://github.com/qgis/QGIS/pull/7372#issuecomment-403172521.

## Checklist

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/server/qgsserverplugins.h [hunk 1]
```diff
@@ -33,6 +33,10 @@ class QgsPythonUtils;
 class SERVER_EXPORT QgsServerPlugins
 {
   public:
+
+    /**
+     * Constructor.
```
Comment (m-kuhn) [inline src/server/qgsserverplugins.h]: What do you think about introducing a test that avoids docstrings like this one ;)

File: src/server/qgsserverrequest.h [hunk 2]
```diff
@@ -90,6 +90,8 @@ class SERVER_EXPORT QgsServerRequest
       */
     QgsServerRequest::Method method() const;
 
+    bool fake() { return true; }
```
Comment (m-kuhn) [inline src/server/qgsserverrequest.h]: What's that for?

**Comment by m-kuhn** [comment]
While I really appreciate improved documentation, I wonder if it's worth introducing "stating the obvious-style" docstrings?

File: src/server/qgsserverrequest.h (hunk 2 above)
Comment (pblottiere) [reply src/server/qgsserverrequest.h]: It was just to test Travis configuration, I'll remove it :)

**Comment by pblottiere** [comment]
> I wonder if it's worth introducing "stating the obvious-style" docstrings?

I don't think so :). 

For now, my purpose is to have a green Travis with the new doc configuration in order to estimate the underlying workload. Then, I'll properly update the documentation.

**Comment by m-kuhn** [comment]
Ok :+1: I'll let you work then :)

**Comment by nyalldawson** [comment]
The intention with the doc test checking constructor docs is that they give useful info ;). The test should already skip any constructors which don't have arguments. So the test is really wanting you to document what these arguments are and the effect they have.

**Comment by pblottiere** [comment]
Arrgghh, content of `\class` is wrong for some classes, and the corresponding group is not found then...

Sorry for the noise :).

**Comment by nyalldawson** [comment]
This is becoming a mega-documentation PR... lovely work!

**Comment by pblottiere** [comment]
Travis is finally happy!

I'll merge this PR in the coming days if I don't have any comments in the meantime.

**Comment by rldhont** [comment]
thanks @pblottiere

**Pull request merged**